sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -9004,4 +9021,25 @@ pi_result _pi_buffer::free() {
   return PI_SUCCESS;
 }
 
+inline pi_result piDeviceTime::get(uint64_t *deviceTime) {
+  if (!initialized) {
+    std::unique_lock{mutex};
+    initialized = true;
+    ZeTimerResolution = device->ZeDeviceProperties->timerResolution;
+    TimestampMaxCount =
+        ((1ULL << device->ZeDeviceProperties->kernelTimestampValidBits) - 1ULL);
+  }
+  uint64_t deviceClockCount, dummy;
+  ZE_CALL(zeDeviceGetGlobalTimestamps,
+          (device->ZeDevice, &dummy, &deviceClockCount));
+  *deviceTime = (deviceClockCount & TimestampMaxCount) * ZeTimerResolution;
+  return PI_SUCCESS;
+}
+
+inline pi_result piDeviceTime::getSubmitTime(pi_event *event) {
```

Comment:
`pi_event` is aleady a pointer, why do you need extra indirection?

Reply:
Removed extra indirection